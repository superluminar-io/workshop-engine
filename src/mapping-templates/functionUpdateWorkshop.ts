export const request = `
  {
    "version" : "2017-02-28",
    "operation" : "UpdateItem",
    "key" : {
      "id": $util.dynamodb.toDynamoDBJson($ctx.args.workshopId),
    },
    "update": {
      "expression" : "SET title = :title, awsAccountId = :awsAccountId, attendees = :attendees",
      "expressionValues": {
        ":title" : $util.dynamodb.toDynamoDBJson($ctx.args.input.title),
        ":awsAccountId" : $util.dynamodb.toDynamoDBJson("$util.defaultIfNullOrEmpty($ctx.args.input.awsAccountId, '')"),
        ":attendees" : $util.dynamodb.toDynamoDBJson($ctx.args.input.attendees)
      }
    }
  }
`;

export const response = `
  $util.toJson($ctx.result)
`;
