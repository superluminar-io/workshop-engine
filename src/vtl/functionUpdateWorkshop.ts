export const request = `
  {
    "version" : "2017-02-28",
    "operation" : "UpdateItem",
    "key" : {
      "id": $util.dynamodb.toDynamoDBJson($ctx.arguments.workshopId)
    },
    "update": {
      "expression" : "SET title = :title, awsAccountId = :awsAccountId, attendees = :attendees",
      "expressionValues": {
        ":title" : $util.dynamodb.toDynamoDBJson($ctx.arguments.input.title),
        ":awsAccountId" : $util.dynamodb.toDynamoDBJson("$util.defaultIfNullOrEmpty($ctx.arguments.input.awsAccountId, '')"),
        ":attendees" : $util.dynamodb.toDynamoDBJson($ctx.arguments.input.attendees)
      }
    }
  }
`;

export const response = `
  $util.toJson($ctx.result)
`;
