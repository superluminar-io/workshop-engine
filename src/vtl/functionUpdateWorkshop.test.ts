import Parser from 'appsync-template-tester';
import { request, response } from './functionUpdateWorkshop';

describe('request mapping', () => {
  describe('input w/ aws account id', () => {
    it('should match snapshot', () => {
      const parser = new Parser(request);

      const input = {
        title: 'Serverless Beginner Workshop',
        awsAccountId: '298097285261',
        attendees: [
          'me@xample.com',
          'you@xample.com',
        ],
      };

      const context = {
        arguments: {
          workshopId: 'beb79a10-437e-48df-96e2-9cd87fdc063f',
          input,
        },
      };

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });

  describe('input w/o aws account id', () => {
    it('should match snapshot', () => {
      const parser = new Parser(request);

      const input = {
        title: 'Serverless Beginner Workshop',
        attendees: [
          'me@xample.com',
          'you@xample.com',
        ],
      };

      const context = {
        arguments: {
          workshopId: 'beb79a10-437e-48df-96e2-9cd87fdc063f',
          input,
        },
      };

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });
});

describe('response mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(response);

    const context = {
      result: {
        any: 'data',
      },
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});
