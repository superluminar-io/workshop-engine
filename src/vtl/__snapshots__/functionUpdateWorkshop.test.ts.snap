// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request mapping input w/ aws account id should match snapshot 1`] = `
Object {
  "key": Object {
    "id": Object {
      "S": "beb79a10-437e-48df-96e2-9cd87fdc063f",
    },
  },
  "operation": "UpdateItem",
  "update": Object {
    "expression": "SET title = :title, awsAccountId = :awsAccountId, attendees = :attendees",
    "expressionValues": Object {
      ":attendees": Object {
        "L": Array [
          Object {
            "S": "me@xample.com",
          },
          Object {
            "S": "you@xample.com",
          },
        ],
      },
      ":awsAccountId": Object {
        "S": "298097285261",
      },
      ":title": Object {
        "S": "Serverless Beginner Workshop",
      },
    },
  },
  "version": "2017-02-28",
}
`;

exports[`request mapping input w/o aws account id should match snapshot 1`] = `
Object {
  "key": Object {
    "id": Object {
      "S": "beb79a10-437e-48df-96e2-9cd87fdc063f",
    },
  },
  "operation": "UpdateItem",
  "update": Object {
    "expression": "SET title = :title, awsAccountId = :awsAccountId, attendees = :attendees",
    "expressionValues": Object {
      ":attendees": Object {
        "L": Array [
          Object {
            "S": "me@xample.com",
          },
          Object {
            "S": "you@xample.com",
          },
        ],
      },
      ":awsAccountId": Object {
        "S": "",
      },
      ":title": Object {
        "S": "Serverless Beginner Workshop",
      },
    },
  },
  "version": "2017-02-28",
}
`;

exports[`response mapping should match snapshot 1`] = `
Object {
  "any": "data",
}
`;
